packages/block-library/src/template-part/edit/index.js

```diff
@@ -254,16 +254,18 @@ export default function TemplatePartEdit( {
 							</ToolbarButton>
 						</BlockControls>
 					) }
-				<InspectorControls group="advanced">
-					<TemplatePartAdvancedControls
-						tagName={ tagName }
-						setAttributes={ setAttributes }
-						isEntityAvailable={ isEntityAvailable }
-						templatePartId={ templatePartId }
-						defaultWrapper={ areaObject.tagName }
-						hasInnerBlocks={ hasInnerBlocks }
-					/>
-				</InspectorControls>
+				{ canUserEdit && (
```

Comment:
There's something that feels off to me here: why are we checking the user rights to edit template part when these changes won't be made in the template part object? they will be made in the container (whether that is a post or template is unknown here)

Reply:
Come to think of it, you are right.

Does this mean that we need additional code like the following to check if the template (not the template part) can be edited?

```javascript
const canUserEditTemplates = select( coreStore ).canUser( 'create', {
	kind: 'postType',
	name: 'wp_template',
} );
```

Reply:
Not really IMO, it feels like a symptom of something else. Do you know why the block is selectable in the first place here? I mean if you can't edit a template, why should you be able to select a block within that template?

Reply:
> Do you know why the block is selectable in the first place here?

From what I can see, #60010 makes it possible to select template parts in the page content focus mode.

Perhaps should we make some changes to the code changed in #60010 to prevent non-admin users from selecting template parts?

Reply:
> Perhaps should we make some changes to the code changed in https://github.com/WordPress/gutenberg/pull/60010 to prevent non-admin users from selecting template parts?

That would be a good intermediary step but to be honest I never liked the fact that you are able to select template parts in "page focus mode", it feels contradictory to me and I'm not sure about the UX improvement.  cc @noisysocks @richtabor